calculator.go

```diff
@@ -1,13 +1,88 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(a, b, c float64) float64 {
+	return a + b + c
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b, c float64) float64 {
+	return (a - b) - c
+}
+
+//Multiply takes two numbers and returns the result of multiplying the second
+// by the first.
+func Multiply(a, b, c float64) float64 {
+	return (a * b) * c
+}
+
+//Divide takes two numbers and returns the result of dividing the second by the first.
+func Divide(a, b, c float64) (float64, error) {
+	if b == 0 || c == 0 {
+		return 0.0, errors.New("divide: divide by zero")
+	}
+	return (a / b) / c, nil
+}
+
+//Sqrt takes a number and returns its square root
+func Sqrt(a float64) (float64, error) {
+	if a < 0 {
+		return 0.0, errors.New("sqrt: negative numbers don't have real square roots")
+	}
+	return math.Sqrt(a), nil
+}
+
+func CalcString(expr string) (float64, error) {
+
+	operators := map[string]string{"multiplication": "*", "division": "/", "addition": "+", "subtraction": "-"}
+
+	var operator string
+
+	for k, v := range operators {
```

Comment:
A slightly easier way to do this (or, at least, one with less code) is to use `fmt.Sscanf`.

Reply:
Yeah, just read its documentation, its definitely what I need. Thank you.